Automattic/jetpack
# Connection: confirm blog ID and access token were saved

## Proposed changes:
* Confirm blog ID and access token were actually saved into the database before proceeding with the connection flow.

### Other information:


## Jetpack product discussion
Discovered here: p1697539842254349-slack-CBG1CP4EN

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Disconnect a Jetpack site.
2. Edit the file `wp-includes/option.php`: find the `update_option()` method and simulate a failure to save by replacing:
`$result = $wpdb->update( $wpdb->options, $update_args, array( 'option_name' => $option ) );`
with:
`$result = false;`
3. Try to connect. Confirm that connection screen displays an error: `An error occurred. Please try again.`.
4. Revert the `wp-includes/option.php` changes.
5. Try to connect again, confirm it goes well.
